components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2080,7 +2116,10 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
                 self.fsm.peer.tag, pending_commit, commit
             );
         }
+        // Clear merge releted data
         self.fsm.peer.pending_merge_state = None;
+        self.fsm.peer.want_rollback_merge_peers.clear();
```

Comment:
Should clear when applying snapshot.

Reply:
After applying a snapshot, `on_ready_rollback_merge` will be called.